tools/transformer/src/transformations/discriminator.js

```diff
@@ -0,0 +1,80 @@
+const {
+  getNameFromYamlPath,
+  getObjectFromYamlPath,
+  getAllObjects,
+  getObjectNameFromReferenceString,
+} = require("../engine/readers");
+
+const { filterObjectProperties } = require("../engine/transformers");
+
+/**
+ * Transforms provided API JSON file using discriminatorOneOf transformation
+ *
+ * @param {*} api OpenAPI JSON File
+ * @returns OpenAPI JSON File
+ */
+function applyDiscriminatorTransformations(api) {
+  const transformations = getAllObjects(api, isTransformable);
+
+  console.info(
+    "# Discriminator transformations: ",
+    transformations.map((e) => e.path)
+  );
+
+  for (let { path } of transformations) {
+    transformDiscriminatorOneOf(path, api);
+  }
+  return api;
+}
+
+// Adds oneOf when discriminator is present
+function transformDiscriminatorOneOf(objectPath, api) {
+  const parentObject = getObjectFromYamlPath(objectPath, api);
+  const parentName = getNameFromYamlPath(objectPath);
+
+  if (!isTransformable(parentObject)) {
+    throw new Error(`Invalid object for AllOf Transformation: ${parentName}`);
+  }
+
+  if (!parentObject.oneOf && parentObject.discriminator) {
+    if (!parentObject.discriminator.mapping) {
+      throw new Error(`OpenAPI object schema contains discriminator but missing oneOf or discriminator mapping.
```

Comment:
This is introducing a form of validation. We should not have this use case in openapi as we would not be able to figure out how to build oneOf. I did ensure that this is the case and introduced this check.